python/cisa_risk_meter/build_cisa_risk_meter.py

```diff
@@ -0,0 +1,334 @@
+        return 0
+    cisa_custom_field = get_custom_field(vulns[0], CISA_CUSTOM_FIELD_NAME)
+    return cisa_custom_field['custom_field_definition_id']
+
+# CVE ID search.
+def search_vulns_for_cve_id(base_url, headers, cve_id):
+    logging.info(f"Searching vulnerabilities for CVE ID {cve_id}")
+
+    # Make a searchable CVE ID.
+    cve_str = cve_id[0:4].upper()
+    if cve_str != "CVE-":
+        print(f"Input is not a CVE, {cve_str}")
+        logging.error(f"Internal error: Expecting CVE, but it was {cve_str}")
+        sys.exit(1)
+
+    cve_id_str = cve_id[4:]
+    
+    # Assemble the Search Vulnerability URL.
+    q_str = f"q=cve:{cve_id_str}"
+    search_url = f"{base_url}vulnerabilities/search?{q_str}"
+    logging.info(f"Vulnerability CVE ID search URL: {search_url}")
+               
+    try: 
+       response = requests.get(search_url, headers=headers)
+    except ConnectionError:
+        logging.warn(f"Connection error, waiting three seconds.  URL: {search_url}")
+        time.sleep(3)
+        response = requests.get(search_url, headers=headers)
+    resp_json = response.json()
+
+    # If too many requests per second, wait a second.
+    if response.status_code == 429:
+        logging.warn(f"Too many search requests, waiting three seconds.  URL: {search_url}")
+        time.sleep(3)
+        response = requests.get(search_url, headers=headers)
+        resp_json = response.json()
+
+    # Check for all other errors.
+    if response.status_code != 200:
+        print(f"Search Vulns Error: {response.status_code} with {search_url}")
+        print_json(resp_json)
+        logging.error(f"Search Vulnerabilities with query string {q_str} API, {search_url}, error: {resp_json}")
+        sys.exit(1)
+    
+    return resp_json['vulnerabilities']
+
+def update_cisa_vulns(base_url, headers, vuln_ids, custom_field_id):
+    if len(vuln_ids) == 0:
+        return
+
+    bulk_update_url = f"{base_url}vulnerabilities/bulk"
+    update_custom_field_id = f"{custom_field_id}"
+    update_data = {
+        "vulnerability_ids": vuln_ids,
+        "vulnerability": {
+            "custom_fields": {
+                update_custom_field_id: "true"
+            }
+        }
+    }
+
+    response = requests.put(bulk_update_url, headers=headers, data=json.dumps(update_data))
+    resp_json = response.json()
+    if response.status_code != 200:
+        print(f"List CISA Catalog Error: {response.status_code} with {bulk_update_url}")
+        logging.error(f"Vulnerability bulk update API, {bulk_update_url}, error: {resp_json}")
+        sys.exit(1)
+    
+    logging.info(f"{len(vuln_ids)} vulnerabilities updated with {CISA_CUSTOM_FIELD_NAME} custom field")
+    
+# Get a risk meter (asset group) by name.
+def get_a_risk_meter(base_url, headers, risk_meter_name):
+    max_pages = API_MAX_PAGES
+    page_num = 1
+    while page_num <= max_pages:
+        list_risk_meters_url = f"{base_url}asset_groups?per_page=100$&page={page_num}"
```

Comment:
It might be better to have a variable for `per_page` amount instead of a hard coded number.

Reply:
Fixed.